rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -134,6 +148,20 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
     return buffer_->use_take_shared_method();
   }
 
+  void
+  provide_serialized_intra_process_message(const SerializedMessage & serialized_message)
```

Comment:
maybe, `provide_ros_massage_from_serialized_message` would be clearer, though a bit verbose.
The opposite version: `provide_serialized_message_from_ros_message` could be added to avoid some duplication in intraprocess manager code.

Reply:
This method should only be available if the message type is a serialized message.
Probably, you can add a SFINAE check above to avoid it to be called in non-serialized message specializations.

Reply:
Though that would not satisfy the base class interface.
Better, if the message type is not serialized, the specialization should throw an error.

Reply:
question: Why is that function called `_serialized_`? Could that not be part of the overload with `provide_intra_process_message(const SerializedMessage & serialized_message)`?